### PR TITLE
[506] Add a location

### PR DESCRIPTION
**Pull request by darokel**
### Context

- https://trello.com/c/ThhZdGGx/506-add-a-location-for-a-provider-in-the-support-console

### Changes proposed in this pull request

- Allow support admin to add locations for a provider

### Guidance to review

Head over to support, login as good ol' Colin, find a provider and add a location for them

### Checklist

- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased master
- [x] Cleaned commit history
- [x] Tested by running locally

### CODE REVIEW

File: app/views/support/locations/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,23 @@
+<%= render PageTitle::View.new(title: "support.providers.locations.new") %>
+
```
Comment (stellenberger) [inline app/views/support/locations/new.html.erb:2]: can we have backlink to provider location index page

File: spec/features/support/providers/locations/managing_locations_spec.rb [hunk 2]
```diff
@@ -54,15 +82,15 @@ def then_should_be_on_the_edit_page_for_the_location
   end
 
   def when_i_edit_the_location_details
-    provider_location_edit_page.location_name.set "Updated location name"
+    provider_location_edit_page.location_form.location_name.set "Updated location name"
```
Comment (stellenberger) [inline spec/features/support/providers/locations/managing_locations_spec.rb:85]:
```suggestion
    provider_location_edit_page.location_form.location_name.set("Updated location name")
```
for consistency

File: app/controllers/support/locations_controller.rb [hunk 3]
```diff
@@ -8,6 +8,21 @@ def index
       redirect_to support_providers_path
     end
 
+    def new
+      provider
```
Comment (stellenberger) [inline app/controllers/support/locations_controller.rb:12]: do we need to set the provider first? 
perhaps worth removing it from `edit` as well if we don't, or into a before action for `new` and `edit`

**Review by stellenberger — APPROVED** [review]
LGTM, just some minor cosmetic things

File: app/controllers/support/locations_controller.rb (hunk 3 above)
Comment (darokel) [reply app/controllers/support/locations_controller.rb:12]: Nope, forgot to remove that. FYI We've opted to use memoised/lazily invoked methods for these sort of situations  - there's an ADR for it here https://github.com/DFE-Digital/teacher-training-api/blob/master/docs/adr/0004-before-action-variables.md.

File: spec/features/support/providers/locations/managing_locations_spec.rb (hunk 2 above)
Comment (darokel) [reply spec/features/support/providers/locations/managing_locations_spec.rb:85]: Excellent eyes

File: app/views/support/locations/new.html.erb (hunk 1 above)
Comment (darokel) [reply app/views/support/locations/new.html.erb:2]: So far all the forms for editing nested resources in support do not have an explicit bank link. There's a cancel link to return back which does the same job. I think if we add it, we'll need to add it to the others too. We can do it in another ticket

File: app/views/support/locations/_form.html.erb [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  <%= f.govuk_text_field :location_name, label: { text: "Name", size: "m" } %>
+
+  <%= f.govuk_text_field(:urn, 
+    label: { text: "Unique Reference Number (URN)", size: "m" }, 
+    hint: 
+      { 
+        text: safe_join(["Find URNs on", " ", tag.a(t("links.get_information_schools"), 
+        href: t("links.get_information_schools"), 
+        target: "_blank",
+        class: "govuk-link")])
```
[the remark this answers is not in the document]
Comment (darokel) [reply app/views/support/locations/_form.html.erb:14]: absolutely

File: app/views/support/locations/_form.html.erb [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+  <%= f.govuk_error_summary %>
+
+  <%= f.govuk_text_field :location_name, label: { text: "Name", size: "m" } %>
+
+  <%= f.govuk_text_field(:urn, 
+    label: { text: "Unique Reference Number (URN)", size: "m" }, 
+    hint: 
+      { 
+        text: safe_join(["Find URNs on", " ", tag.a(t("links.get_information_schools"),
```
[the remark this answers is not in the document]
Comment (darokel) [reply app/views/support/locations/_form.html.erb:11]: good spot, this was definitely poor copy pasta

**Review by mbocc — APPROVED** [review]
I can adit, edit, and delete a location as an Admin user - all looks good, thank you!

**Pull request merged**